test/util/transaction_test_util.cpp

```diff
@@ -44,6 +45,11 @@ void RandomWorkloadTransaction::RandomUpdate(Random *generator) {
     }
     updates_[updated] = update;
   }
+  // TODO(Tianyu): Hardly efficient, but will do for testing.
+  if (test_object_->wal_on_) {
+    auto *record = txn_->StageWrite(nullptr, updated, initializer);
+    TERRIER_MEMCPY(record->Delta(), update, update->Size());
```

Comment:
Is record ever used? I don't think the function StageWrite() has any side effect.

Reply:
its memory content would have changed in the memcpy